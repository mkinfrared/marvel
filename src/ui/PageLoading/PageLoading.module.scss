@import "styles/colors";
@import "styles/vars";

.PageLoading {
  align-items: center;
  animation: animate 250ms linear;
  background-color: rgba(0, 0, 0, 0.2);
  display: flex;
  height: calc(100vh - #{$header-height});
  justify-content: center;
  width: 100%;
}

@keyframes animate {

  0% {
    transform: translateY(100%);
  }

  100% {
    transform: translateY(0%);
  }
}
