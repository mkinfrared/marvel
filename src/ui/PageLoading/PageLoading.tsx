import React, { useEffect } from "react";

import Spinner from "ui/Spinner";

import css from "./PageLoading.module.scss";

const PageLoading = () => {
  useEffect(() => {
    document.body.style.overflow = "hidden";

    document.body.style.height = "100vh";

    return () => {
      document.body.style.overflow = "initial";

      document.body.style.height = "initial";
    };
  }, []);

  return (
    <div className={css.PageLoading}>
      <Spinner size={2} />
    </div>
  );
};

export { PageLoading };

export default React.memo(PageLoading);
