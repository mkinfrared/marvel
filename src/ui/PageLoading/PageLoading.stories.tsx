import { Meta, Story } from "@storybook/react/types-6-0";
import React from "react";

import Header from "ui/AppBar";

import { PageLoading } from "./PageLoading";

export default {
  title: "UI/PageLoading",
  component: PageLoading
} as Meta;

const Template: Story = () => (
  <>
    <Header />
    <PageLoading />
  </>
);

const Primary = Template.bind({});

Primary.args = {};

export { Primary };
