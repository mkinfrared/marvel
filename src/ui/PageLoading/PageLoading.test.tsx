import { render } from "@testing-library/react";
import React from "react";

import { PageLoading } from "./PageLoading";

describe("<PageLoading />", () => {
  const Component = PageLoading;

  it("should be defined", () => {
    const { container } = render(<Component />);

    expect(container).toBeDefined();
  });

  it("match the snapshot", () => {
    const { container } = render(<Component />);

    expect(container).toMatchSnapshot();
  });
});
