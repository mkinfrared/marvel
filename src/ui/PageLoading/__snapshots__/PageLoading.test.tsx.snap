// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageLoading /> match the snapshot 1`] = `
<div>
  <div
    class="PageLoading"
  >
    <div
      class="Spinner"
      style="width: 50px; height: 50px;"
    />
  </div>
</div>
`;
